### PR TITLE
tests: subsys: fs: move fs_api to new ztest API

### PR DESCRIPTION
**Pull request by yinfangw**
Move fs_api to new ztest API.

Signed-off-by: Yinfang Wang <yinfang.wang@intel.com>

### CODE REVIEW

File: tests/subsys/fs/fs_api/src/main.c [hunk 1]
```diff
@@ -65,35 +45,30 @@ static void fs_teardown(void)
...
+	test_file_read();
+	test_file_seek();
+	test_file_truncate();
+	test_file_close();
+	test_file_sync();
+	test_file_rename();
+	test_file_stat();
+	test_file_unlink();
+	test_unmount();
+	test_mount_flags();
```
Comment (yperess) [inline tests/subsys/fs/fs_api/src/main.c:71]: This turns all the tests into a single test. Can you split it up?

**Review by yperess — CHANGES_REQUESTED** [review]

File: tests/subsys/fs/fs_api/src/main.c (hunk 1 above)
Comment (yinfangw) [reply tests/subsys/fs/fs_api/src/main.c:71]: @yperess  https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/subsys/fs/fs_api/src/main.c#L62-L63 said that the order of test cases is critical, one case depend on the case before it.

**Review by mengxianglinx — DISMISSED** [review]

File: tests/subsys/fs/fs_api/src/main.c (hunk 1 above)
Comment (yinfangw) [reply tests/subsys/fs/fs_api/src/main.c]: @yperess please review again.

**Comment by yinfangw** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/49666 had updated the test case, close this one.

**Pull request closed without merging**